### PR TITLE
Feature/pokemon

### PR DESCRIPTION
**Pull request by Sbusta**
angular app

### CODE REVIEW

File: src/app/app.component.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.mat-simple-snackbar{
```
Comment (hackeo1) [inline src/app/app.component.css]: space before {

File: src/app/app.component.ts [hunk 2]
```diff
@@ -5,6 +5,6 @@ import { Component } from '@angular/core';
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
-export class AppComponent {
-  title = 'pokemon-app';
+export class AppComponent{
+  title = 'angular-pokemon';
```
Comment (hackeo1) [inline src/app/app.component.ts]: remove unused variables

File: src/app/app.module.ts [hunk 3]
```diff
@@ -1,17 +1,54 @@
...
+    AppRoutingModule,
+    BrowserAnimationsModule,
+    CommonModule,
+    MaterialModule,
+    HttpClientModule,
+    RouterModule.forRoot(routes),
+    EffectsModule.forRoot([]),
+    StoreModule.forRoot({}, {}),
+    PokemonsModule,
+    StoreDevtoolsModule.instrument({ name:'Talos',maxAge: 25, logOnly: environment.production }),
```
Comment (hackeo1) [inline src/app/app.module.ts]: space after ,

File: src/app/core/localstore.service.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import { Injectable } from "@angular/core";
+import { MatSnackBar, MatSnackBarHorizontalPosition, MatSnackBarVerticalPosition } from "@angular/material/snack-bar";
+import { BehaviorSubject } from "rxjs";
+import { Favorite } from "../shared/models/pokemon.model";
```
Comment (hackeo1) [inline src/app/core/localstore.service.ts]: use single quote in ts files

File: src/app/core/localstore.service.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LocalstoreService{
+  initialFavorites :Favorite[] = [{path: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png", name:'bulbasaur'},
+                                  {path: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png", name:'charmander'},
+                                  {path: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/7.png", name:'squirtle'}];
```
Comment (hackeo1) [inline src/app/core/localstore.service.ts]: use single quote

File: src/app/core/localstore.service.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  favorites = new BehaviorSubject(this.initialFavorites);
+  constructor( private _snackBar: MatSnackBar){
+    localStorage.setItem('favorites',JSON.stringify(this.initialFavorites))
```
Comment (hackeo1) [inline src/app/core/localstore.service.ts]: doesn't this override on every reload?

Comment (hackeo1) [reply src/app/core/localstore.service.ts]: space after ,

File: src/app/core/localstore.service.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    }
+  }
+  
+  getFavorites() {
+    return JSON.parse(localStorage.getItem('favorites') || '{}');
+  }
+
+  deleteFavorite(name:string){
+    let temporalFavorites = this.getFavorites();
+    let index = temporalFavorites.findIndex((x:Favorite) => x.name === name);
```
Comment (hackeo1) [inline src/app/core/localstore.service.ts]: don't use single character variables

File: src/app/core/pokemons.service.ts [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+  constructor(private http: HttpClient) { }
+
+  getPokemons(): Observable<Pokemon[]> {
+    let off = this.page * 20;
+    let lim = 20;
+    let querry = `${this.pokemonApiUrl}?offset=${off}&limit=${lim}`;
+    this.page++;
+    return this.http.get<ApiResp>(querry)
+    .pipe(
+      tap(data => console.log('data',data)),
```
Comment (hackeo1) [inline src/app/core/pokemons.service.ts]: remove unnecesary logs

Comment (hackeo1) [reply src/app/core/pokemons.service.ts]: or add a configuration to disable them

File: src/app/home/home.component.css [hunk 9]
```diff
@@ -0,0 +1,11 @@
+.pokemon-img{
+  image-rendering: pixelated;
+}
+.carousel-cell{
+  text-align: center;
+}
+.carousel{
+  margin-top: 5%;
+  font-weight: 700;
+  font-size: 1.5rem;
```
Comment (hackeo1) [inline src/app/home/home.component.css]: use variables

File: src/app/home/home.component.html [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+  objectFit="contain"
+  [cellWidth]="'100%'" 
+  [autoplay]="true"
+  [dots]="true"
+  [loop]="true"
+  [height]=400
+  [cellsToShow]=3>
+  <div *ngFor="let item of $favorites |async" class="carousel-cell">
+    {{item.name | uppercase}}
+    <img class="pokemon-img"  src="{{item.path}}">
```
Comment (hackeo1) [inline src/app/home/home.component.html]: don't use interpolation ({{}}) use input instead

File: src/app/home/home.component.ts [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+    this.$favorites = this.local.favorites;
+  }
+
+  ngOnInit(): void {
+    
+  }
+}
+
+
+
```
Comment (hackeo1) [inline src/app/home/home.component.ts]: remove extra lines, only one is required at the end

File: src/app/home/home.component.ts [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+export class HomeComponent implements OnInit {
+  $favorites:Observable<Favorite[]>;
+
+  constructor(private store: Store<State>, private local: LocalstoreService) { 
+    this.$favorites = this.local.favorites;
+  }
+
+  ngOnInit(): void {
+    
+  }
```
Comment (hackeo1) [inline src/app/home/home.component.ts]: remove unused functions

File: src/app/material/material.module.ts [hunk 13]
```diff
@@ -0,0 +1,21 @@
+import { NgModule } from '@angular/core';
+import { MatButtonModule } from '@angular/material/button'
+import { MatTabsModule } from '@angular/material/tabs';
+import {MatInputModule} from '@angular/material/input';
+import {MatToolbarModule} from '@angular/material/toolbar';
+import {MatCardModule} from '@angular/material/card';
```
Comment (hackeo1) [inline src/app/material/material.module.ts]: add line after imports

File: src/app/pokemons/chart/chart.component.html [hunk 14]
```diff
@@ -0,0 +1,13 @@
+<div>
+  <div>
+    <div style="display: block">
```
Comment (hackeo1) [inline src/app/pokemons/chart/chart.component.html]: don't use inline styles

File: src/app/pokemons/chart/chart.component.ts [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+  
+  public barChartOptions: ChartOptions = {
+    responsive: true,
+    scales: { xAxes: [{}], 
+              yAxes: [{
+                ticks: {
+                  min: 0,
+                  suggestedMax: 110,
+                }
+              }] },
```
Comment (hackeo1) [inline src/app/pokemons/chart/chart.component.ts]: fix indentation

File: src/app/pokemons/chart/chart.component.ts [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+import * as pluginDataLabels from 'chartjs-plugin-datalabels';
+import { Label } from 'ng2-charts';
+
+@Component({
+  selector: 'app-chart',
+  templateUrl: './chart.component.html',
+  styleUrls: ['./chart.component.css']
+})
+export class ChartComponent implements OnInit {
+  @Input() stats:ChartDataSets[]=[{ data: [65, 59, 80, 81, 56, 55, 40], label: 'Series A' }];
```
Comment (hackeo1) [inline src/app/pokemons/chart/chart.component.ts]: add spaces

File: src/app/pokemons/chart/chart.component.ts [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+
+  public barChartLegend = true;
+
+  public barChartPlugins = [pluginDataLabels];
+
+  constructor() { }
+
+  ngOnInit(): void {
+    
+  }
```
Comment (hackeo1) [inline src/app/pokemons/chart/chart.component.ts]: remove unused functions

File: src/app/pokemons/chart/chart.component.ts [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+
+  public chartClicked({ event, active }: { event: MouseEvent, active: {}[] }): void {
+    console.log(event, active);
+  }
+
+  public chartHovered({ event, active }: { event: MouseEvent, active: {}[] }): void {
+    console.log(event, active);
+  }
```
Comment (hackeo1) [inline src/app/pokemons/chart/chart.component.ts]: remove loggers

File: src/app/pokemons/compare/compare.component.css [hunk 19]
```diff
@@ -0,0 +1,47 @@
+.pokemon-img{
+  display: inline-block;
+  width: 25vw;
+
```
Comment (hackeo1) [inline src/app/pokemons/compare/compare.component.css]: remove extra line

File: src/app/pokemons/compare/compare.component.css [hunk 20]
```diff
@@ -0,0 +1,47 @@
+.pokemon-img{
+  display: inline-block;
+  width: 25vw;
```
Comment (hackeo1) [inline src/app/pokemons/compare/compare.component.css]: avoid the use of vw to give width, how does this look on mobile?

File: src/app/pokemons/compare/compare.component.html [hunk 21]
```diff
@@ -0,0 +1,61 @@
...
+        {{pokemonB.height/10}} m
+      </span>
+      <span class="description-value">
+        {{pokemonA.weight/10}} m
+      </span>
+      <span class="description-title">
+        Weight:
+      </span>
+      <span class="description-value">
+        {{pokemonB.weight/10}} m
```
Comment (hackeo1) [inline src/app/pokemons/compare/compare.component.html]: avoid the use of calculations inside interpolation, they are done every time it renders

File: src/app/pokemons/compare/compare.component.ts [hunk 22]
```diff
@@ -0,0 +1,72 @@
...
+import * as PokemonsActions from '../state/pokemons.actions';
+import { Stats } from 'src/app/shared/models/pokemon.model';
+@Component({
+  selector: 'app-compare',
+  templateUrl: './compare.component.html',
+  styleUrls: ['./compare.component.css']
+})
+export class CompareComponent implements OnInit, OnDestroy {
+  pokemonA: Stats = {name: '', height: 0, weight: 0, sprites: {front_default:''}, abilities: [{ability:{name:''}}], types: [{type:{name:''}}], stats: 
+      [{base_stat: 0 },{base_stat: 0},{base_stat: 0},{base_stat: 0},{base_stat: 0},{base_stat: 0}]};
```
Comment (hackeo1) [inline src/app/pokemons/compare/compare.component.ts]: improve indentation, use a constant to initialize the values

File: src/app/pokemons/details/details.component.css [hunk 23]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+.stats{
+  display: flex;
+  flex-flow: wrap;
+  justify-content: space-between;
+}
+
+.favorite{
+  background-color: yellow;
```
Comment (hackeo1) [inline src/app/pokemons/details/details.component.css]: don't use named colors

File: src/app/pokemons/details/details.component.css [hunk 24]
```diff
@@ -0,0 +1,76 @@
...
+  .description-container-list{
+    margin: 4%;
+  }
+
+  .description-container-list{
+    margin: 4%;
+    width: 40%;
+  }
+
+  .description-container-list ul{
```
Comment (hackeo1) [inline src/app/pokemons/details/details.component.css]: avoid the use of tags for styling

File: src/app/pokemons/details/details.component.html [hunk 25]
```diff
@@ -0,0 +1,67 @@
...
+          <span class="description-value">
+            {{pokemon.height/10}}m
+          </span>
+        </span>
+        <span class="description-container">
+          <span class ="description-title">
+            Weight:
+          </span>
+          <span class="description-value">
+            {{pokemon.weight/10}}kg
```
Comment (hackeo1) [inline src/app/pokemons/details/details.component.html]: don't add calculations in interpolations

File: src/app/pokemons/details/details.component.ts [hunk 26]
```diff
@@ -0,0 +1,125 @@
...
+  errorMessage: string = '';
+  pokemon: Stats = {name: '', height: 0, weight: 0, sprites: {front_default:''}, abilities: [{ability:{name:''}}], types: [{type:{name:''}}], stats: 
+  [{base_stat: 0 },{base_stat: 0},{base_stat: 0},{base_stat: 0},{base_stat: 0},{base_stat: 0}]};;
+  description: Details = {flavor_text_entries:[{flavor_text:''}]};
+  stats: ChartDataSets[] = [];
+  gender: string = '';
+  descriptionText: string = '';
+  pokemonA:Stats={name: '', height: 0, weight: 0, sprites: {front_default:''}, abilities: [{ability:{name:''}}], types: [{type:{name:''}}], stats: 
+  [{base_stat: 0 },{base_stat: 0},{base_stat: 0},{base_stat: 0},{base_stat: 0},{base_stat: 0}]};;
+  isFavorite:boolean= false;
```
Comment (hackeo1) [inline src/app/pokemons/details/details.component.ts]: space missing

File: src/app/pokemons/details/details.component.ts [hunk 27]
```diff
@@ -0,0 +1,125 @@
...
+})
+export class DetailsComponent implements OnInit, AfterContentInit, OnDestroy {
+  errorMessage: string = '';
+  pokemon: Stats = {name: '', height: 0, weight: 0, sprites: {front_default:''}, abilities: [{ability:{name:''}}], types: [{type:{name:''}}], stats: 
+  [{base_stat: 0 },{base_stat: 0},{base_stat: 0},{base_stat: 0},{base_stat: 0},{base_stat: 0}]};;
+  description: Details = {flavor_text_entries:[{flavor_text:''}]};
+  stats: ChartDataSets[] = [];
+  gender: string = '';
+  descriptionText: string = '';
+  pokemonA:Stats={name: '', height: 0, weight: 0, sprites: {front_default:''}, abilities: [{ability:{name:''}}], types: [{type:{name:''}}], stats:
```
Comment (hackeo1) [inline src/app/pokemons/details/details.component.ts]: check indentation and spaces

File: src/app/pokemons/details/details.component.ts [hunk 28]
```diff
@@ -0,0 +1,125 @@
...
+  [{base_stat: 0 },{base_stat: 0},{base_stat: 0},{base_stat: 0},{base_stat: 0},{base_stat: 0}]};;
+  isFavorite:boolean= false;
+
+  constructor(@Inject(MAT_DIALOG_DATA) public data: any, private store: Store<State>, public dialog: MatDialog, private local:LocalstoreService) {
+    
+  }
+
+  checkFavorite(name:string){
+    let fav=this.local.getFavorites();
+    let containsName = fav.some( (favorite:Favorite) => favorite.name === name)
```
Comment (hackeo1) [inline src/app/pokemons/details/details.component.ts]: is fav a list? use more descriptive names

File: src/app/pokemons/details/details.component.ts [hunk 29]
```diff
@@ -0,0 +1,125 @@
...
+    );
+    
+
+    this.store.select(getPokemonA).subscribe(
+      pokemon => this.pokemonA = pokemon!
+    );
+  }
+  
+  getDescription():string{
+    let obj = this.description.flavor_text_entries.find((element: any) => element.language.name === 'en');
```
Comment (hackeo1) [inline src/app/pokemons/details/details.component.ts]: don't use any

File: src/app/pokemons/pokemoncard/pokemoncard.component.ts [hunk 30]
```diff
@@ -0,0 +1,30 @@
...
+  constructor(public dialog: MatDialog, private pokemonsService: PokemonsService, private store: Store<State>) { }
+  detailsUrl='https://pokeapi.co/api/v2/pokemon-species/'
+  selectPokemon(){
+    let Url=`${this.detailsUrl}${this.pokemon.url.split('/')[6]}/`
+    let poke:Pokemon ={...this.pokemon,imgUrl:this.imgUrl};
+    this.store.dispatch(PokemonsActions.setSelectedPokemon({pokemon:poke}));
+    this.store.dispatch(PokemonsActions.loadPokemon({url:this.pokemon.url}));
+    this.store.dispatch(PokemonsActions.loadPokemonDetails({url:Url}))
+    const dialogRef = this.dialog.open(DetailsComponent, {panelClass:'details-dialog'});
+  }
```
Comment (hackeo1) [inline src/app/pokemons/pokemoncard/pokemoncard.component.ts]: use new lines to organize code, the previous code is also missing spaces

File: src/app/pokemons/pokemonlist/pokemonlist.component.css [hunk 31]
```diff
@@ -0,0 +1,36 @@
...
+  flex-flow: row wrap;
+}
+
+.pokeball{
+  display: flex;
+  justify-content: center;
+}
+
+.snack{
+  color: blue;
```
Comment (hackeo1) [inline src/app/pokemons/pokemonlist/pokemonlist.component.css]: don't use named colors

File: src/app/pokemons/pokemonlist/pokemonlist.component.html [hunk 32]
```diff
@@ -0,0 +1,9 @@
+<div class="pokemon-list-container">
+<app-pokemoncard *ngFor="let pokemon of pokemons$ | async; index as i" [pokemon]="pokemon" imgUrl="{{ pokemon.url.split('/')[6] | formatImgUrl}}" class="example-card"></app-pokemoncard>
+</div>
+<section *ngIf="!filtering" waIntersectionObserver waIntersectionThreshold="0.5" class="pokeball">
+    <div (waIntersectionObservee)="onIntersection($event)">
+        <img src="../../assets/simple_pokeball.gif" alt="pokeball-gif" class="pokeball-gif">
```
Comment (hackeo1) [inline src/app/pokemons/pokemonlist/pokemonlist.component.html]: don't use relative paths for resources

File: src/app/pokemons/pokemonlist/pokemonlist.component.ts [hunk 33]
```diff
@@ -0,0 +1,97 @@
...
+  ngOnInit(): void {
+    this.store.dispatch(PokemonsActions.loadPokemons());
+  }
+
+  ngAfterContentInit(): void {
+    this.filtering = false;
+  }
+
+  onIntersection(event:IntersectionObserverEntry[]){
+    event.map((inter) => {
```
Comment (hackeo1) [inline src/app/pokemons/pokemonlist/pokemonlist.component.ts]: is this an event? it looks like it is an array of entries

File: src/app/pokemons/pokemonlist/pokemonlist.component.ts [hunk 34]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  ngOnDestroy(): void {
+    this.isFiltering$.unsubscribe();
+    this.compare$.unsubscribe();
+    this.dialogClosed$.unsubscribe();
+    this.pokemonA$.unsubscribe();
+  }
+
+  compare = false;
```
Comment (hackeo1) [inline src/app/pokemons/pokemonlist/pokemonlist.component.ts]: organize your class attributes in a single group

File: src/app/shared/format-img-url.pipe.ts [hunk 35]
```diff
@@ -0,0 +1,11 @@
+import { Pipe, PipeTransform } from "@angular/core";
+
+@Pipe({
+  name: 'formatImgUrl'
+})
+export class FormatImgUrlPipe implements PipeTransform {
+  imageUrl = 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/';
```
Comment (hackeo1) [inline src/app/shared/format-img-url.pipe.ts]: urls should be centralized

**Review by hackeo1 — CHANGES_REQUESTED** [review]
- add spaces after colon and commas
- use single quotes in ts files
- add space before {
- remove unit tests if they don't work
- indentation in ts and css files should be 2 spaces
- template must use double quotes for attributes
- avoid the use of % for paddings and margins

**Review by hackeo1 — CHANGES_REQUESTED** [review]
I'm not able to see the file changes, it looks like there is a problem with the PR

**Comment by Sbusta** [comment]
> I'm not able to see the file changes, it looks like there is a problem with the PR

humm, maybe closing the PR and open another one?

**Pull request closed without merging**